packages/amesos2/src/Amesos2_Lapack_def.hpp

```diff
@@ -289,16 +289,16 @@ namespace Amesos2 {
       // typedef Util::get_ccs_helper<MatrixAdapter<Matrix>,
       //        scalar_type, global_ordinal_type, global_size_type> ccs_helper;
     typedef Util::get_ccs_helper_kokkos_view<MatrixAdapter<Matrix>,
-        host_value_type_array, host_ordinal_type_array, host_size_type_array> ccs_helper;
+        host_value_type_array, host_ordinal_type_array, host_ordinal_type_array> ccs_helper;
     if ( is_contiguous_ == true ) {
       ccs_helper::do_get(this->matrixA_.ptr(),
                          nzvals_view_, rowind_view_, colptr_view_,
-                         nnz_ret, ROOTED, SORTED_INDICES, 0);
+                         nnz_ret, ROOTED, ARBITRARY, this->rowIndexBase_);
     }
     else {
       ccs_helper::do_get(this->matrixA_.ptr(),
                          nzvals_view_, rowind_view_, colptr_view_,
-                         nnz_ret, CONTIGUOUS_AND_ROOTED, SORTED_INDICES, 0);
+                         nnz_ret, CONTIGUOUS_AND_ROOTED, ARBITRARY, this->rowIndexBase_);
```

Comment:
Totally aside from the changes in this PR, but I happened to be here looking:
* Based on the `if (is_contiguous_)` condition, is the use of `ROOTED` vs  `CONTIGUOUS_AND_ROOTED` reversed in the two calls?
* (unclear to me whether this matters) Does switching from `SORTED_INDICES` to `ARBITRARY` have negative performance implications, such that the code should conditionalize that for when it's necessary?

Reply:
@PhilMiller :

Re: point 1 - no, though the convention is a bit counter-intuitive (at least I found a bit confusing); e.g. CONTIGUOUS_AND_ROOTED means to hit the code path to make contiguous and rooted, which is the correct path when the GIDs are not contiguous

Reply:
That reversed convention, of the argument being a request rather than an affirmation, would perhaps be worth noting in a comment.

Reply:
Or, the enumerated values for that argument could be renamed to indicative that they're directives for what the code should do, not preconditions that it will rely upon